src/Common/src/TypeSystem/IL/Stubs/PInvokeILEmitter.cs

```diff
@@ -128,6 +128,14 @@ private MethodIL EmitIL()
 
                 callsiteSetupCodeStream.Emit(ILOpcode.call, emitter.NewToken(nativeMethod));
             }
+            
+            // if the SetLastError flag is set in DllImport, call the PInvokeMarshal.SaveLastWin32Error so that last error can be used later 
+            // by calling PInvokeMarshal.GetLastWin32Error
+            if ((importMetadata.Attributes & PInvokeAttributes.SetLastError) == PInvokeAttributes.SetLastError)
```

Comment:
While you are on it, it may be nice to call ClearLastError before the call as well for compatibility with CoreCLR.